### ISSUE TITLE
[GeoNature] Paramétrage filtres statuts

### ISSUE THREAD

**Issue by okristo** [cbn-alpin/silene-geonature#182]
Pour le paramétrage des filtres de statuts, dernière version du tableau mentionnant les textes à prendre en compte :
[Selection_statuts_silene-GK.xlsx](https://github.com/cbn-alpin/sinp-paca-tickets/files/8098260/Selection_statuts_silene-GK.xlsx)

Possibilité d'ajouter l'entrée "directives européennes" ? Cf. onglet spécifique dans le tableau ci-dessus.

**Comment by jpm-cbna** [comment]
A priori, dans ton tableur, je peux me baser sur le champ CD_DOC qui est aussi présent dans les tables de GeoNature.

Le CD_DOC ne permet pas d'identifier tous les textes car il y en a 9 653 différent pour 11 344 textes mais cela me semble le seul identifiant utilisable pour l'instant.

Par contre, dans le tableur, est ce que je peux récupérer la liste des CD_DOC depuis l'onglet "tous_statuts_applicables_pac" ?
Est ce que c'est bien un onglet qui regroupe tous les statuts présents dans les autres onglets ?
Si c'est bien le cas, je compte dedans que 15 textes sans CD_DOC. J'essaierai de les retrouver manuellement.

**Comment by okristo** [comment]
Non, l'onglet tous-statuts-applicables-PACA reprend l'ensemble des textes qui pourraient s'appliquer à PACA mais tous n'ont pas été retenus.
Ceux qui sont retenus sont dans les onglets suivants (protection_nationale, protection_regionale,...). Je les ai compilés dans le fichier suivant
EDIT (mise à jour doc intégrant les listes rouges nationales, europ, et mondiale) : 
[CD_DOC_selection.xlsx](https://github.com/cbn-alpin/sinp-paca-tickets/files/8197991/CD_DOC_selection.xlsx)

**Comment by jpm-cbna** [comment]
Ok, parfait. Du coup, il y en a très peu.

**Comment by jpm-cbna** [comment]
Pour la configuration des filtres qui s'affiche dans l'interface, la configuration par défaut correspond aux besoins du SINP PACA. Il n'y a donc rien à changer.

La requête permettant de déselectionner tous les textes que nous ne voulons pas est :
```sql
UPDATE taxonomie.bdc_statut_text
SET enable = false
WHERE cd_doc NOT IN (366749, 901, 738, 758, 763, 625, 633, 3561, 643, 713, 716, 730, 731, 
    703, 694, 694, 732, 733, 174768, 174769, 174770, 195368, 268129, 
    268409, 146732, 145082, 196448, 158248, 755, 756, 358269, 358270, 
    160321, 275396, 31345, 138062, 31343, 300831, 138065, 87486, 165208, 
    87625, 31341, 87619, 138063, 144173, 220350, 321049, 208629, 87484, 
    146311, 88261, 300212, 146310, 31346, 249369, 138064);
```

Un [script présent dans le dépôt sinp-paca-data](https://github.com/cbn-alpin/sinp-paca-data/blob/master/db-geonature/data/sql/04_disable_status_text.sql) contient la requête ci-dessus.

**Comment by okristo** [comment]
Je viens de mettre à jour le doc listant les statuts à prendre en compte (cf edit de mon post précédent)

**Comment by helenechauvin** [comment]
Bonjour, je reprends ce ticket. Pourquoi n'a-t-on pas pu intégrer les statuts des directives européennes? Nous avons des demandes en ce sens. La base utilisée le permet-elle? (directive habitat faune flore (avec les différentes annexes) et divective oiseaux (idem)).
Je parle bien du requêtage dans les filtres taxonomiques avancés, pas dans l'export du fichier statuts

**Comment by okristo** [comment]
Les CD_DOC correspondant (755 et 756) sont bien listés dans la sélection mais effectivement, nous ne les avons pas fait apparaître dans les filtres (un oubli a priori).
il faudrait ajouter un champ spécifique "directives européennes" ramenant ces deux CD_DOC.

**Comment by okristo** [comment]
CD_DOC cités plus haut (755 et 756) à ajouter
